LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/CreateDraftPayRunUseCase.cs

```diff
@@ -28,29 +27,48 @@ public CreateDraftPayRunUseCase(IPayRunGateway payRunGateway, IQueueService payr
 
         public async Task CreateDraftPayRun(DraftPayRunCreationDomain draftPayRunCreationDomain)
         {
-            var draftPayRunCount = await _payRunGateway.GetDraftPayRunCount(draftPayRunCreationDomain.Type);
-            if (draftPayRunCount > 0)
-                throw new ApiException($"A Pay Run with draft status for {draftPayRunCreationDomain.Type.ToDescription()} already exists!");
-
+            var recurringPayRunTypes = new[] { PayrunType.DirectPayments, PayrunType.ResidentialRecurring };
             ValidateDraftPayRun(draftPayRunCreationDomain);
 
+            var unApprovedPayRunExists = await _payRunGateway.CheckExistsUnApprovedPayRunAsync(draftPayRunCreationDomain.Type);
+            if (unApprovedPayRunExists)
+                throw new ApiException($"Operation not allowed. There exists a pay run of the same type that is not approved", HttpStatusCode.PreconditionFailed);
+
             draftPayRunCreationDomain.Status = PayrunStatus.Draft;
-            draftPayRunCreationDomain.StartDate = await _payRunGateway.GetDateOfLastPayRun(draftPayRunCreationDomain.Type);
+
+            if (recurringPayRunTypes.Contains(draftPayRunCreationDomain.Type))
+            {
+                draftPayRunCreationDomain.StartDate = await _payRunGateway.GetEndDateOfLastPayRun(draftPayRunCreationDomain.Type);
+            }
+            else
+            {
+                draftPayRunCreationDomain.StartDate = draftPayRunCreationDomain.PaidFromDate.GetValueOrDefault();
+            }
 
             var payrun = draftPayRunCreationDomain.ToEntity();
 
             await _payRunGateway.CreateDraftPayRun(payrun);
             await _payrunsQueue.Send(payrun.Id);
         }
 
-        private static void ValidateDraftPayRun(DraftPayRunCreationDomain draftPayRunCreationDomain)
+        private static void ValidateDraftPayRun(DraftPayRunCreationDomain creationDomain)
         {
-            if (draftPayRunCreationDomain.PaidUpToDate > DateTimeOffset.Now)
+            if (creationDomain.PaidUpToDate.Date > DateTimeOffset.Now.Date)
```

Comment:
Where is written this logic? Payrun PaidUpToDate can be later than today?

Reply:
Just to be sure. Datepicker, or postman can allow future selection. Unless pay run end date can be in the future :)

Reply:
There is no need to check EndDate or PaidUpToDate. User can be able to create payrun request for endOfMonth on middleOfMonth.

Reply:
Done